pkg/controller/devopsproject/devopsproject_controller.go

```diff
@@ -254,6 +254,10 @@ func (c *Controller) syncHandler(key string) error {
 				ns := c.generateNewNamespace(project)
 				ns, err := c.client.CoreV1().Namespaces().Create(ns)
 				if err != nil {
+					// recreate devops with generateName if admin namespace is conflict
+					if errors.IsAlreadyExists(err) && project.GenerateName != "" {
+						return c.recreateDevOpsProject(project)
```

Comment:
It will cause the data of users to lose if there is the same name of DevOps.

Reply:
This is the logic of DevOps project creation, regenerate DevOps project ID if admin namespace is conflicted. DevOps project associated with the Jenkins project(admin namespace) will not be recreated.

Reply:
If you delete devopproject  obj，the project in jenkins will be deleted

https://github.com/kubesphere/kubesphere/blob/3c8fbe39b971108c81d41b683caf8c9b45437963/pkg/controller/devopsproject/devopsproject_controller.go#L310

Reply:
https://github.com/kubesphere/kubesphere/blob/3c8fbe39b971108c81d41b683caf8c9b45437963/pkg/controller/devopsproject/devopsproject_controller.go#L208-L253

Admin namespace is empty, there is no namespace association,  because the admin namespace creation conflict.

Reply:
We need to make sure that the admin namespace is created properly, regenerate project ID is needed.